src/components/shared/Button/InviteLinkButton.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect, useCallback } from 'react';
```

Comment:
I have been thinking more about where to place components I think it may be better to organize components by their intended use instead of if they are already used in multiple places or not.

1. Group single purpose components by their relationship to other components. Example, many of the components in common/ColonyHome. For example: NewActionButton is good example as the intention is that there would only be 1 action button in the CDapp (although I feel it may make more sense if this button was in a subfolder of ColonyHomeLayout).
- Pros
    - Easy to see the hierarchy of components
    - Easy to determine where to put non-shared components
- Cons
    - The component hierarchy will be missing shared components
    - Lots of moving components around to different folders if they become shared

2. Group all components with the intention of being generalist/shareable components in a shared folder. Since these components are meant to be re-used they always have the potential of being shared across multiple components.
- Pros
    - A component’s location does not need to move after creation
    - This way of thinking is more likely to lead to components with a single responsibility and shareable from the start
    - Components that are designed to be shared instead of used for only one place in the CDapp should lead to a more consistent design and less work to update the design
- Cons
    - Could be more work to generalize components

Overall, I think we will have reasons to build both types of components but I think it will lead to more maintainability if we try to use shared components first before using specific purpose components. For the InviteLinkButton, I think it may have potential to be re-used in other places in the CDapp and it doesn’t necessary feel designed just for this one use.